tilings/strategies/sliding.py

```diff
@@ -0,0 +1,276 @@
+            _gp_reverse(c),
+        )
+
+    @classmethod
+    def rotate90(cls, r: int) -> "_AdditionalMaps":
+        return _AdditionalMaps(
+            _tiling_rotate90,
+            _tiling_rotate90_inverse,
+            _gp_rotate90(r),
+            _gp_rotate90_inverse(r),
+        )
+
+    @classmethod
+    def rotate90_and_reverse(cls, r: int) -> "_AdditionalMaps":
+        return _AdditionalMaps(
+            _tiling_rotate90_and_reverse,
+            _tiling_rotate90_and_reverse_inverse,
+            _gp_rotate90_and_reverse(r),
+            _gp_rotate90_and_reverse_inverse(r),
+        )
+
+
+class SlidingStrategy(DisjointUnionStrategy[Tiling, GriddedPerm]):
+    """
+    A class for a specific slidding strategy. The init probably gets the index of both
+    column you slidding.
+    """
+
+    def __init__(
+        self,
+        av_12_column: int,
+        av_123_column: int,
+        col_info: DefaultDict[int, DefaultDict[int, Dict[GriddedPerm, List[int]]]],
+        maps: _AdditionalMaps = _AdditionalMaps(),
+    ):
+        super().__init__(possibly_empty=False)
+        self.av_12 = av_12_column
+        self.av_123 = av_123_column
+        self.col_info = col_info
+        self.maps = maps
+
+    def decomposition_function(self, tiling: Tiling) -> Tuple[Tiling]:
+        """Return the slidded tiling if it slides, otherwise None."""
+        return (
+            self.maps.t_inv(
+                slide_column(
+                    self.maps.t_map(tiling), self.av_12, self.av_123, self.col_info
+                )
+            ),
+        )
+
+    def formal_step(self) -> str:
+        return f"slide {self.av_123} through {self.av_12}"
+
+    def backward_map(
+        self,
+        tiling: Tiling,
+        gps: Tuple[Optional[GriddedPerm], ...],
+        children: Optional[Tuple[Tiling, ...]] = None,
+    ) -> Iterator[GriddedPerm]:
+        if self.av_12 < self.av_123:
+            yield from (
+                self.maps.g_inv(
+                    gp_slide_inverse(self.maps.g_map(gp), self.av_12, self.av_123)
+                )
+                for gp in gps
+                if gp is not None
+            )
+        else:
+            yield from (
+                self.maps.g_inv(gp_slide(self.maps.g_map(gp), self.av_123, self.av_12))
+                for gp in gps
+                if gp is not None
+            )
+
+    def forward_map(
+        self,
+        tiling: Tiling,
+        gp: GriddedPerm,
+        children: Optional[Tuple[Tiling, ...]] = None,
+    ) -> Tuple[GriddedPerm]:
+        if self.av_123 < self.av_12:
+            return (
+                self.maps.g_inv(gp_slide(self.maps.g_map(gp), self.av_123, self.av_12)),
+            )
+        return (
+            self.maps.g_inv(
+                gp_slide_inverse(self.maps.g_map(gp), self.av_12, self.av_123)
+            ),
+        )
+
+    @classmethod
+    def from_dict(cls, d: dict) -> "SlidingStrategy":
+        raise NotImplementedError()
```

Comment:
We'll need to implement the json methods before accepting. I'll add some things to test files that you need to make sure pass.